0000-miner-update.md

```diff
@@ -0,0 +1,199 @@
+
+* How are we soliciting feedback on this HIP from these stakeholders?
+
+Feedback will be gathered by sharing this HIP.
+
+# Detailed Explanation
+[detailed-explanation]: #detailed-explanation
+
+Helium's private nextgate build system builds Hotspot firmware, signs it and
+uploads it to a public AWS S3 bucket for downloading. The nexthaul OTA server
+provides metadata about where to find and how to verify available Hotspot
+firmware releases. The Helium Blockchain Engineering team updates this release
+information inside the helium/coop repo and pushes changes to Heroku for
+deployment to the existing fleet of Hotspots. Hotspots use a self-signed public
+SSL certificate included in their firmware to establish that the ota.helium.com
+server on the other end is in fact Helium's and not some bad actor's.
+
+The nextgate repo currently builds miner as a separate software package that is
+then installed to a target root file system for inclusion in the Hotspot's
+embedded Linux image. The entire system image is signed by Helium but the
+individual miner software artifacts currently are not. The firmware image is
+signed so that consumer Hotspots can verify that an over-the-air update
+originated from Helium before committing it to flash storage and rebooting.
+
+The monolithic firmware updates that nextgate/nexthaul performs are too
+dangerous for Hotspot owners to administer themselves. The chances of an end
+user deploying a homemade firmware update that accidentally bricks their
+Hotspots is simply too high. What we want is a way for Hotspot owners to only
+upgrade the miner portion of their devices' firmware so that their Hotspots
+remain online and continue to receive hardware-specific updates (Linux kernel,
+gateway-config, LoRa packet_forwarder) only from Helium.
+
+The simplest solution is to install miner and all its dependencies (RocksDB,
+libsodium, ebus) as an image overlay on top of a Hotspot's root file system.
+This extraction step would happen at boot time once Linux has started and the
+immutable root file system has been loaded into memory. The end result is
+similar to how web applications are deployed as tarballs for extracting to and
+running on a newly-launched AWS EC2 instance. A new minerhaul OTA server will
+need to be open sourced and client-side scripts will need to be added to
+nextgate to download and apply miner updates.
+
+Another solution is to package miner releases in Docker containers for
+deployment to and execution on Hotspots. Instead of pointing their Hotspots at
+alternative minerhaul servers Hotspot owners would instead point them at
+alternative Docker registries. A miner package maintainer uses Docker push
+commands to publish new miner images to her registry. Hotspots use Docker pull
+commands to fetch new miner images from their designated registries. A Docker
+runtime and client-side scripts will need to be added to nextgate for automatic
+teardown and setup of miner images on the Hotspots.
```

Comment:
From @Vagabond 

> Propose we proceed with Docker until we find a solid reason it's unsuitable.

Reply:
That's a good point. A user can still roll back to a previously working version of the Hotspot firmware by button-booting. However, a better fix is for not being able to pair over BLE is for the GATT server to read a Hotspot's name from the ECC rather than ask miner for it.

Reply:
Sounds like a plan.